WooCommerce/Classes/ViewRelated/Dashboard/Settings/Domains/DomainSelectorView.swift

```diff
@@ -79,88 +58,112 @@ struct DomainSelectorView: View {
 
     @State private var isWaitingForDomainSelectionCompletion: Bool = false
 
+    @FocusState private var textFieldIsFocused: Bool
+
     init(viewModel: DomainSelectorViewModel) {
         self.viewModel = viewModel
     }
 
     var body: some View {
-        VStack(alignment: .leading, spacing: 0) {
-            ScrollView {
-                VStack(alignment: .leading) {
-                    // Header label.
-                    Text(Localization.subtitle)
-                        .foregroundColor(Color(.secondaryLabel))
-                        .bodyStyle()
-                        .padding(.horizontal, Layout.defaultHorizontalPadding)
+        ScrollView {
+            VStack(alignment: .leading, spacing: 0) {
+                // Header label.
+                Text(Localization.subtitle)
+                    .foregroundColor(Color(.secondaryLabel))
+                    .bodyStyle()
+                    .padding(.horizontal, Layout.defaultHorizontalPadding)
+                    .padding(.top, 16)
 
-                    // Search text field.
-                    SearchHeader(text: $viewModel.searchTerm,
-                                 placeholder: Localization.searchPlaceholder,
-                                 customizations: .init(backgroundColor: .clear,
-                                                       borderColor: .separator,
-                                                       internalHorizontalPadding: 21,
-                                                       internalVerticalPadding: 12))
+                Spacer()
+                    .frame(height: 30)
 
-                    // Results header.
-                    Text(Localization.suggestionsHeader)
+                // Search text field.
+                SearchHeader(text: $viewModel.searchTerm,
+                             placeholder: Localization.searchPlaceholder,
+                             customizations: .init(backgroundColor: .clear,
+                                                   borderColor: .separator,
+                                                   internalHorizontalPadding: 21,
+                                                   internalVerticalPadding: 12,
+                                                   iconSize: .init(width: 14, height: 14)))
+                .focused($textFieldIsFocused)
+
+                if viewModel.searchTerm.isEmpty {
```

Comment:
Nit: IMO, to cover the different scenarios, we could use a State enum in the view model with cases like empty, loading, error and suggestions and use a switch case here.

Reply:
Thanks for your suggestion! I like this state enum approach, and updated in https://github.com/woocommerce/woocommerce-ios/pull/8165/commits/1b685f46f537ccffbc5842deb96c9ab1cf8c8acb if you'd like to take a look 🙂 (I kept the main conditional logic the same but now it's in the view model and testable)

Reply:
Thanks for the update! It looks great. 🎉